libs/contrib/Data/Elem/Extra.idr

```diff
@@ -0,0 +1,61 @@
+module Data.Elem.Extra
+
+import Data.List
+import Data.List.Elem
+
+%default total
+
+||| Convenience lemma showing that prepending nil to xs is just xs.
+public export
+appendNilLeftNeutral : (xs : List a) -> ([] ++ xs) = xs
+appendNilLeftNeutral [] = Refl
+appendNilLeftNeutral (x :: xs) = Refl
+
+
+||| Proof that an element is still inside a list if we append to it.
+public export
+elemAppLeft : (xs, ys : List a) -> (prf : Elem x xs) -> Elem x (xs ++ ys)
+elemAppLeft (x :: xs) ys Here = Here
+elemAppLeft (x :: xs) ys (There prf2) = There $ elemAppLeft xs ys prf2
+
+
+||| Proof that an element is still inside a list if we prepend to it.
+public export
+elemAppRight :  (xs, ys : List a) -> (prf : Elem x xs) -> Elem x (ys ++ xs)
```

Comment:
I'd rather keep `xs` and `ys` in the same order as in the resulting type.
Otherwise it's always pretty confusing to remember to swap the arguments
around when using `elemAppLeft` vs. `elemAppRight`.

Reply:
Sounds good to me, I'll make the change and try again!